src/libraries/System.Diagnostics.Process/src/System/Diagnostics/Process.Windows.cs

```diff
@@ -460,8 +460,8 @@ private unsafe bool StartWithCreateProcess(ProcessStartInfo startInfo)
             //    * CreateProcess allows you to redirect all or none of the standard IO handles, so we use
             //      GetStdHandle for the handles that are not being redirected
 
-            StringBuilder commandLine = BuildCommandLine(startInfo.FileName, StartInfo.Arguments);
-            Process.AppendArguments(commandLine, StartInfo.ArgumentList);
+            StringBuilder commandLine = BuildCommandLine(startInfo.FileName);
+            startInfo.AppendArguments(commandLine);
```

Comment:
Nit: I'd probably want to include this line in the `BuildCommandLine` method (and accordingly update its signature to something like `StringBuilder BuildCommandLine(ProcessStartInfo)`

Reply:
good idea,  will do that